message.go

```diff
@@ -699,8 +703,16 @@ func newMessageTag(tag reflect.StructTag) messageTag {
 }
 
 func emptyMessageElement(rv reflect.Value) bool {
+
 	switch rv.Kind() {
 
+	// These types are never considered empty since their zero values
+	// can be considered valid.
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
```

Comment:
Maybe this function should default to returning `false`, so all these types don't need a `case`. I think I would prefer to have `case`'s for types with "truly empty" zero values that should be marked as empty, e.g. `string`.